lib/record_store/zone.rb

```diff
@@ -272,6 +273,26 @@ def validate_provider_can_handle_zone_records
       end
     end
 
+    def validate_zone_record_not_shadowed
+      nameserver_records = records
```

Comment:
bit of a nitpick but maybe `nameserver_fqdns` since this isn't `Record` objects but rather just fqdns in strings?